ape_hardhat/provider.py

```diff
@@ -262,10 +283,12 @@ def project_folder(self) -> Path:
 
     @property
     def uri(self) -> str:
-        if not self.port:
-            raise HardhatProviderError("Can't build URI before `connect()` is called.")
-
-        return f"http://127.0.0.1:{self.port}"
+        if self.config.host:
+            if "https" not in self.config.host or "http" not in self.config.host:
+                self.config.host = "http://" + self.config.host  # type: ignore
```

Comment:
instead of changing `self.config.host`, i wonder if we can just do this straight on `_host` ?